src/Bridges/bridge_optimizer.jl

```diff
@@ -1241,6 +1244,42 @@ function MOI.supports(
     )
 end
 
+function _set_substituted(
+    b::AbstractBridgeOptimizer,
+    attr::MOI.ConstraintSet,
+    ci::MOI.ConstraintIndex{F,S},
+    value,
+) where {F,S}
+    if !(value isa S)
+        throw(ArgumentError("Invalid type when setting ConstraintSet."))
```

Comment:
It doesn't seem to be the right place for this. MOIU.Model also does not have the right error:
```julia
julia> model = MOI.Utilities.Model{Float64}()
MOIU.Model{Float64}

julia> b = MOI.Bridges.full_bridge_optimizer(model, Float64)
MOIB.LazyBridgeOptimizer{MOIU.Model{Float64}}
with 0 variable bridges
with 0 constraint bridges
with 0 objective bridges
with inner model MOIU.Model{Float64}

julia> x = MOI.add_variable(b)
MathOptInterface.VariableIndex(1)

julia> MOI.add_constraint(model, x, MOI.EqualTo(0.0))
MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.EqualTo{Float64}}(1)

julia> MOI.set(b, MOI.ConstraintSet(), x, MOI.LessThan(1.0))
ERROR: ArgumentError: `supports` is not defined for MathOptInterface.ConstraintSet(), it is only defined for attributes such that `is_copyable` returns `true`.
Stacktrace:
 [1] supports_fallback(#unused#::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.Model{Float64}}, attr::MathOptInterface.ConstraintSet, #unused#::Type{MathOptInterface.VariableIndex})
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:249
 [2] supports(model::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.Model{Float64}}, attr::MathOptInterface.ConstraintSet, args::Type)
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:224
 [3] throw_set_error_fallback(model::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.Model{Float64}}, attr::MathOptInterface.ConstraintSet, index::MathOptInterface.VariableIndex, value::MathOptInterface.LessThan{Float64}; error_if_supported::MathOptInterface.SetAttributeNotAllowed{MathOptInterface.ConstraintSet})
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:506
 [4] throw_set_error_fallback(model::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.Model{Float64}}, attr::MathOptInterface.ConstraintSet, index::MathOptInterface.VariableIndex, value::MathOptInterface.LessThan{Float64})
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:506
 [5] set(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.Model{Float64}}, ::MathOptInterface.ConstraintSet, ::MathOptInterface.VariableIndex, ::MathOptInterface.LessThan{Float64})
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:481
 [6] top-level scope
   @ REPL[14]:1

julia> MOI.set(model, MOI.ConstraintSet(), x, MOI.LessThan(1.0))
ERROR: ArgumentError: `supports` is not defined for MathOptInterface.ConstraintSet(), it is only defined for attributes such that `is_copyable` returns `true`.
Stacktrace:
 [1] supports_fallback(#unused#::MathOptInterface.Utilities.Model{Float64}, attr::MathOptInterface.ConstraintSet, #unused#::Type{MathOptInterface.VariableIndex})
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:249
 [2] supports(model::MathOptInterface.Utilities.Model{Float64}, attr::MathOptInterface.ConstraintSet, args::Type)
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:224
 [3] throw_set_error_fallback(model::MathOptInterface.Utilities.Model{Float64}, attr::MathOptInterface.ConstraintSet, index::MathOptInterface.VariableIndex, value::MathOptInterface.LessThan{Float64}; error_if_supported::MathOptInterface.SetAttributeNotAllowed{MathOptInterface.ConstraintSet})
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:506
 [4] throw_set_error_fallback(model::MathOptInterface.Utilities.Model{Float64}, attr::MathOptInterface.ConstraintSet, index::MathOptInterface.VariableIndex, value::MathOptInterface.LessThan{Float64})
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:506
 [5] set(::MathOptInterface.Utilities.Model{Float64}, ::MathOptInterface.ConstraintSet, ::MathOptInterface.VariableIndex, ::MathOptInterface.LessThan{Float64})
   @ MathOptInterface ~/.julia/dev/MathOptInterface/src/attributes.jl:481
 [6] top-level scope
   @ REPL[15]:1
```

Reply:
Why not add a `throw_set_error_fallback` method so that we fix this for all model at the same time ?

Reply:
This is wrong because you're using `x` instead of the constraint index.

Reply:
```Julia
julia> model = MOI.Utilities.Model{Float64}()
MOIU.Model{Float64}

julia> b = MOI.Bridges.full_bridge_optimizer(model, Float64)
MOIB.LazyBridgeOptimizer{MOIU.Model{Float64}}
with 0 variable bridges
with 0 constraint bridges
with 0 objective bridges
with inner model MOIU.Model{Float64}

julia> x = MOI.add_variable(b)
MathOptInterface.VariableIndex(1)

julia> c = MOI.add_constraint(model, x, MOI.EqualTo(0.0))
MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.EqualTo{Float64}}(1)

julia> MOI.set(b, MOI.ConstraintSet(), c, MOI.LessThan(1.0))
ERROR: ArgumentError: Invalid type when setting ConstraintSet.
Stacktrace:
 [1] _set_substituted(b::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.Model{Float64}}, attr::MathOptInterface.ConstraintSet, ci::MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.EqualTo{Float64}}, value::MathOptInterface.LessThan{Float64})
   @ MathOptInterface.Bridges ~/.julia/dev/MathOptInterface/src/Bridges/bridge_optimizer.jl:1254
 [2] set(b::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.Model{Float64}}, attr::MathOptInterface.ConstraintSet, ci::MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.EqualTo{Float64}}, value::MathOptInterface.LessThan{Float64})
   @ MathOptInterface.Bridges ~/.julia/dev/MathOptInterface/src/Bridges/bridge_optimizer.jl:1304
 [3] top-level scope
   @ REPL[11]:1
```